crypto/initthread.c

```diff
@@ -121,6 +123,32 @@ init_get_thread_local(CRYPTO_THREAD_LOCAL *local, int alloc, int keep)
     return hands;
 }
 
+#else
+static THREAD_EVENT_HANDLER **
+init_get_thread_local_ex(OSSL_LIB_CTX *ctx, int alloc, int keep)
```

Comment:
We currently call `init_get_thread_local()` with two forms:
  - `init_get_thread_local(ctx, 1, 0);`
  - `init_get_thread_local(ctx, 0, 0);`

it feels like arg2 (`keep`) is always zero. can the argument be removed completely?

Reply:
ossl_ctx_thread_start  calls  init_get_thread_local(ctx, 1, 0) in the non-fips case.

Reply:
I see where the confusion is coming from here. The `keep` argument at `init_get_thread_local_ex()` is always 0, therefore, I think it can be removed for `_ex()` variant.

Reply:
To be honest I don't like this _swiss army knife_ kind of function. I would prefer to have more functions with proper name. For example:
   init_get_thread_local(&x, 0, 0); => get_thread_local(&x);
   init_get_thread_local(&x, 1, .); => create_thread_local(&x);
   init_get_thread_local(&x, ., 1); => destroy_thread_local(&x);
the 1,0 arguments make reading the code lot harder. This is just suggestion I don't insist on.

Reply:
> To be honest I don't like this _swiss army knife_ kind of function. 

Or if you want a single function they have flags with macros for their values (or enums if internal) so it is clearer.

init_get_thread_local(&x, OSSL_THREAD_LOCAL_GET); 
init_get_thread_local(&x, OSSL_THREAD_LOCAL_CREATE); or _INIT?
init_get_thread_local(&x, OSSL_THREAD_LOCAL_DESTROY);

It also allows for more "operations" to be added without adding more functions or needing to add "ex" functions etc.

Reply:
I'm with @Sashan on this one - I'd just create 3 functions and use them as appropriate. These are internal functions so there is no problem with creating new one or changing semantics of existing one.